code/core/src/manager/components/sidebar/TestingModule.tsx

```diff
@@ -271,11 +270,17 @@ export const TestingModule = ({
             }}
           >
             <Content ref={contentRef}>
-              {Object.values(registeredTestProviders).map((state) => {
-                const { render: Render } = state;
+              {Object.values(registeredTestProviders).map((registeredTestProvider) => {
+                const { render: Render, id } = registeredTestProvider;
+                if (!Render) {
+                  console.warn(
+                    `No render function found for test provider with id '${id}', skipping...`
+                  );
+                  return null;
+                }
                 return (
-                  <TestProvider key={state.id} data-module-id={state.id}>
-                    {Render ? <Render {...state} /> : <LegacyRender {...state} />}
+                  <TestProvider key={id} data-module-id={id}>
```

Comment:
Do we need to add an error boundary here? I'm wondering whether an addon can break the entire testing module if it has some buggy code there